velox/core/PlanNode.h

```diff
@@ -1524,6 +1524,54 @@ class CrossJoinNode : public PlanNode {
   const RowTypePtr outputType_;
 };
 
+/// Represents inner/outer nested loop joins. Translates to an
+/// exec::NestedLoopJoinProbe and exec::CrossJoinBuild (NestedLoopJoin shares
+/// build operator/bridge with CrossJoin). A separate pipeline is produced for
+/// the build side when genreating exec::Operators. Nested loop join supports
+/// non-equal joins. Expressions specified in joinCondition are evaluated on
+/// every combination of left/right tuple, to emit result.
+class NestedLoopJoinNode : public PlanNode {
+ public:
+  NestedLoopJoinNode(
+      const PlanNodeId& id,
+      JoinType joinType,
+      TypedExprPtr joinCondition,
```

Comment:
Let's remove CrossJoinNode and use NestedLoopJoinNode with nullptr for 'joinCondition'.

Reply:
Have removed as suggested.
1. For compatibility, I kept interfaces that were used to create `CrossJoinNode`, only updated them to create null-joinCondition `NestedLoopJoinNode`:
- `PlanBuilder::crossJoin`
- `PlanNodePtr toVeloxPlan(::duckdb::LogicalCrossProduct& logicalCrossProduct, ...)`

2. As `CrossJoinProbe` is the subset of `NestedLoopJoinProbe` in terms of functionality, maybe we can remove it as well?

Reply:
@usurai Thank you for the follow-up. Let's remove references to CrossJoinNode/Operator completely.

Reply:
@mbasmanova I have replaced `CrossJoinProbe` with `NestedLoopJoinProbe` and renamed `CrossJoin{Bridge/Build}` to `NestedLoopJoin{Bridge/Build}`.